packages/venia-ui/lib/components/App/localeProvider.js

```diff
@@ -61,13 +64,13 @@ const LocaleProvider = props => {
         }
     };
 
-    if (loading) return <LoadingIndicator global={true} />;
+    if (!messages) return <LoadingIndicator global={true} />;
```

Comment:
I wonder which is better:

a) Showing a loading indicator or
b) Not showing a loading indicator, ever, opting to have a potential "flash" when the translated strings come back from `fetchLocale`.

It would be really cool if we could read `defaultMessage` strings from some bundled file that agencies could customize depending on their own default language...

Reply:
That's comparing develop.pwa-venia.com to this PR, with this aiming to remove the render blocking of the loading indicator. Rendering the loading indicator means the context isn't getting rendered, including its children (the rest of the app).

The second part you outlined is exactly what this change does; while Apollo is still waiting on its response, we default to english and render the app immediately. If the response comes back and it's the same as the default, nothing re-renders.

I think the ideal solution here is to apply this logic to all stores; we can precache `storeConfig` at build time about all stores and immediately load the {StoreConfig.locale} language pack. If we don't want to require rebuilds when config changes, we can leave the query in place that will true up with the network to validate the precached config. I think we've been hinting at needing this solution for awhile, and would be another application to prevent unnecessary re-renders.

Reply:
@tjwiebell a setting to have storeConfig pre-loaded would be great.